x/bank/keeper/virtual.go

```diff
@@ -0,0 +1,224 @@
+package keeper
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// SendCoinsFromAccountToModuleVirtual sends coins from account to a virtual module account.
+func (k BaseSendKeeper) SendCoinsFromAccountToModuleVirtual(
+	ctx context.Context, senderAddr sdk.AccAddress, recipientModule string, amt sdk.Coins,
+) error {
+	recipientAcc := k.ak.GetModuleAccount(ctx, recipientModule)
+	if recipientAcc == nil {
+		return errorsmod.Wrapf(sdkerrors.ErrUnknownAddress, "module account %s does not exist", recipientModule)
+	}
+
+	return k.SendCoinsToVirtual(ctx, senderAddr, recipientAcc.GetAddress(), amt)
+}
+
+// SendCoinsFromModuleToAccountVirtual sends coins from account to a virtual module account.
+func (k BaseSendKeeper) SendCoinsFromModuleToAccountVirtual(
```

Comment:
maybe we do add sub and add callback in https://github.com/crypto-org-chain/cosmos-sdk/blob/0e712ba89bbff2f8c0735681ac993019b6547ec7/x/bank/keeper/send.go#L212 to reuse here

Reply:
reuse more code, right?

Reply:
maybe split SendCoins into smaller methods.

Reply:
yea

Reply:
done